docs/DeveloperGuide.md

```diff
@@ -756,9 +780,117 @@ testers are expected to do more *exploratory* testing.
+
+    3. Test case: `list t/activity s/START`<br>
+       Expected: GoMedic shows both activities, but the activity with id `A002` is shown before the activity with id `A001`.
+
+    4. Test case: `list t/activity s/START p/ALL`<br>
+       Expected: GoMedic shows both activities, but the activity with id `A002` is shown before the activity with id `A001`.
+
+    5. Test case: `list t/activity p/HEHE`<br>
+       Expected: The feedback box displays an error message about the parameter supplied.
+    
+    6. Other valid `list t/activity` commands to try: First, add an activity whose date for the `START_TIME` and `END_TIME` 
+       corresponds to date that the user tests the `list t/activity` command. Then, run `list t/activity s/ID p/TODAY`. <br>
+       Expected: GoMedic shows a table of activities that includes the activity mentioned above.
+     
+    7. Other invalid `list t/activity` commands to try: `list t/activites`, `list t/activity s/HOHO` (invalid parameter supplied), `...` <br>
+       Expected: Feedback box displays error message indicating an invalid command / parameter constraints violations.
+
+
+### Listing all patients
+
+1. Listing patients in GoMedic
+
+    1. **Prerequisites**: Clear all existing patients in GoMedic using `clear t/patient` command. 
+       Add 2 new patients by running `add t/patient n/John Smith p/98765432 a/45 b/AB+ g/M h/175 w/70 m/heart failure m/diabetes` and 
+       `add t/patient n/Joan Lim p/12345678 a/30 b/A- g/F h/165 w/45 m/high blood pressure` in order. 
+       Conduct the following tests in sequential order. 
+
+    2. Test case: `list t/patient`<br>
+       Expected: GoMedic shows a table with 2 patients, one with id `P001` and another with id `P002`.
+
+    3. Test case: `list t/patient extra parameters supplied here`<br>
+       Expected: GoMedic shows a table with 2 patients, one with id `P001` and another with id `P002`, as it ignores the 
+       extra parameters supplied.
+     
+    4. Other invalid `list t/patient` commands to try: `list t/patients` <br>
+       Expected: Feedback box displays error message indicating an invalid command.
+
+### Listing all doctors
+
+1. Listing doctors in GoMedic
+
+    1. **Prerequisites**: Clear all existing doctors in GoMedic using `clear t/doctor` command. 
+       Add 2 new doctors by running `add t/doctor n/John Smith p/98765432 de/Cardiology` and 
+       `add t/doctor n/Tom Hill p/12345678 de/Radiology` in order. 
+       Conduct the following tests in sequential order. 
+
+    2. Test case: `list t/doctor`<br>
+       Expected: GoMedic shows a table with 2 doctors, one with id `D001` and another with id `D002`.
+
+    3. Test case: `list t/doctor extra parameters supplied here`<br>
+       Expected: GoMedic shows a table with 2 doctors, one with id `D001` and another with id `D002`, as it ignores the 
+       extra parameters supplied.
+     
```

Comment:
Wah, this is very detailed, I thought at first just combine this into # list patient, doctor or activity but sure